--- mjpegtools-1.8.0/lavtools/Makefile.am.orig	2005-06-16 19:09:37.000000000 +0200
+++ mjpegtools-1.8.0/lavtools/Makefile.am	2006-12-25 12:51:09.583398899 +0100
@@ -40,21 +40,32 @@
 	-release $(LT_RELEASE)
 
 liblavfile_la_SOURCES = editlist.c lav_io.c avilib.c
-liblavfile_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBDV_CFLAGS) $(JPEG_CFLAGS) $(LIBQUICKTIME_CFLAGS)
+liblavfile_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBDV_CFLAGS) $(LIBQUICKTIME_CFLAGS)
 liblavfile_la_LDFLAGS = $(LAV_ALL_LIB_OPTS) \
-	$(LIBQUICKTIME_LIBS) $(JPEG_LIBS) $(LIBDV_LIBS)
+	$(LIBQUICKTIME_LIBS) $(LIBDV_LIBS)
+liblavfile_la_LIBADD = $(top_builddir)/utils/libmjpegutils.la
 
 liblavjpeg_la_SOURCES = jpegutils.c
 liblavjpeg_la_CPPFLAGS = $(AM_CPPFLAGS) $(JPEG_CFLAGS)
 liblavjpeg_la_LDFLAGS = $(LAV_ALL_LIB_OPTS) $(JPEG_LIBS)
+liblavjpeg_la_LIBADD = $(top_builddir)/utils/libmjpegutils.la
 
 liblavrec_la_SOURCES = liblavrec.c audiolib.c frequencies.c
 liblavrec_la_CPPFLAGS = $(AM_CPPFLAGS) # so automake doesn't complain about audiolib & frequencies
 liblavrec_la_LDFLAGS = $(LAV_ALL_LIB_OPTS)
+liblavrec_la_LIBADD = liblavfile.la liblavjpeg.la \
+	$(top_builddir)/utils/libmjpegutils.la
+
+if HAVE_X
+  LAVPLAY_X_LIBS = $(X_LIBS) -lX11
+endif
 
 liblavplay_la_SOURCES = liblavplay.c audiolib.c
 liblavplay_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBDV_CFLAGS) $(X_CFLAGS)
 liblavplay_la_LDFLAGS = $(LAV_ALL_LIB_OPTS)
+liblavplay_la_LIBADD = liblavfile.la liblavjpeg.la \
+	$(top_builddir)/utils/libmjpegutils.la \
+	$(SDL_LIBS) $(LAVPLAY_X_LIBS)
 
 # *********************************************************************
 # The tools themselves
@@ -138,10 +149,6 @@
 multiblend_flt_SOURCES = multiblend.flt.c
 multiblend_flt_LDADD = $(LIBMJPEGUTILS)
 
-if HAVE_X
-  LAVPLAY_X_LIBS = $(X_LIBS) -lX11
-endif
-
 lavplay_SOURCES = lavplay.c
 lavplay_LDADD = $(LIBMJPEGUTILS) \
 	liblavplay.la liblavjpeg.la liblavfile.la \
