--- mjpegtools-1.8.0/mpeg2enc/mpeg2encoder.hh.orig	2006-01-19 20:59:50.000000000 +0100
+++ mjpegtools-1.8.0/mpeg2enc/mpeg2encoder.hh	2006-01-19 20:59:58.000000000 +0100
@@ -22,7 +22,6 @@
  *
  */
 
-#include "config.h"
 #include <stdio.h>
 #include "mpeg2encoptions.hh"
 #include "encoderparams.hh"
--- mjpegtools-1.8.0/mpeg2enc/encodertypes.h.orig	2006-01-19 21:10:55.000000000 +0100
+++ mjpegtools-1.8.0/mpeg2enc/encodertypes.h	2006-01-19 21:11:01.000000000 +0100
@@ -24,7 +24,6 @@
  *
  */
 
-#include "config.h"
 
 
 
--- mjpegtools-1.8.0/mpeg2enc/quantize_ref.h.orig	2006-01-19 21:11:17.000000000 +0100
+++ mjpegtools-1.8.0/mpeg2enc/quantize_ref.h	2006-01-19 21:11:23.000000000 +0100
@@ -23,7 +23,6 @@
 #ifndef _QUANTIZE_H
 #define _QUANTIZE_H
 
-#include "config.h"
 #include "mjpeg_types.h"
 
 struct QuantizerWorkSpace;
--- mjpegtools-1.8.0/mpeg2enc/mpeg2encoptions.hh.orig	2006-01-19 21:15:37.000000000 +0100
+++ mjpegtools-1.8.0/mpeg2enc/mpeg2encoptions.hh	2006-01-19 21:15:47.000000000 +0100
@@ -22,7 +22,6 @@
  *
  */
 
-#include <config.h>
 #include "mpeg2encparams.h"
 #include "mpegconsts.h"
 
--- mjpegtools-1.8.0/mpeg2enc/mpeg2encparams.h.orig	2006-01-19 21:15:18.000000000 +0100
+++ mjpegtools-1.8.0/mpeg2enc/mpeg2encparams.h	2006-01-19 21:15:22.000000000 +0100
@@ -22,7 +22,6 @@
  *
  */
 
-#include <config.h>
 
 
 
--- mjpegtools-1.8.0/mpeg2enc/encoderparams.hh.orig	2006-01-19 21:27:17.000000000 +0100
+++ mjpegtools-1.8.0/mpeg2enc/encoderparams.hh	2006-01-19 21:27:23.000000000 +0100
@@ -54,7 +54,6 @@
 
 
 
-#include "config.h"
 #include "mjpeg_types.h"
 #include "syntaxconsts.h"
 
--- mjpegtools-1.8.0/mpeg2enc/Makefile.am.orig	2006-01-19 21:52:19.000000000 +0100
+++ mjpegtools-1.8.0/mpeg2enc/Makefile.am	2006-01-19 21:53:28.000000000 +0100
@@ -75,7 +75,7 @@
 	$(mpeg2enc_REF) \
 	$(SIMD_INLINE)
 
-noinst_HEADERS = channel.hh 	mpeg2syntaxcodes.h 	quantize_precomp.h 	simd.h \
+noinst_HEADERS = channel.hh 	quantize_precomp.h 	simd.h \
 		tables.h 	$(mpeg2enc_noinst_header_REF)
 
 libmpeg2encpp_includedir = $(pkgincludedir)/mpeg2enc
@@ -89,6 +89,7 @@
 	mpeg2encoder.hh \
 	mpeg2encoptions.hh \
 	mpeg2encparams.h \
+	mpeg2syntaxcodes.h \
 	picture.hh \
 	picturereader.hh \
 	quantize.hh \
