diff -Nru mjpegtools-1.6.1.orig/mplex/multplex.cc mjpegtools-1.6.1/mplex/multplex.cc
--- mjpegtools-1.6.1.orig/mplex/multplex.cc	2002-04-29 14:36:00.000000000 +0000
+++ mjpegtools-1.6.1/mplex/multplex.cc	2003-06-09 20:38:18.000000000 +0000
@@ -3,6 +3,7 @@
 #include <math.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <assert.h>
 
 #include <mjpeg_types.h>
 #include <mjpeg_logging.h>
diff -Nru mjpegtools-1.6.1.orig/mplex/stillsstream.cc mjpegtools-1.6.1/mplex/stillsstream.cc
--- mjpegtools-1.6.1.orig/mplex/stillsstream.cc	2002-04-18 04:32:02.000000000 +0000
+++ mjpegtools-1.6.1/mplex/stillsstream.cc	2003-06-09 21:37:31.000000000 +0000
@@ -22,6 +22,7 @@
 
 
 #include <format_codes.h>
+#include <assert.h>
 
 #include "stillsstream.hh"
 #include "interact.hh"
diff -Nru mjpegtools-1.6.1.orig/mplex/systems.cc mjpegtools-1.6.1/mplex/systems.cc
--- mjpegtools-1.6.1.orig/mplex/systems.cc	2002-04-29 14:36:01.000000000 +0000
+++ mjpegtools-1.6.1/mplex/systems.cc	2003-06-09 21:08:04.000000000 +0000
@@ -4,6 +4,7 @@
 #include <string.h>
 #include <sys/param.h>
 #include <sys/stat.h>
+#include <assert.h>
 #include "systems.hh"
 #include "mplexconsts.hh"
 
