--- mjpegtools-1.6.0/configure.in.orig	Mon May  6 10:33:25 2002
+++ mjpegtools-1.6.0/configure.in	Sat Dec 21 19:51:36 2002
@@ -58,6 +58,7 @@
 AC_C_BIGENDIAN
 AC_C_CONST
 AC_C_INLINE
+AM_PROG_AS
 
 dnl Initialize libtool
 AM_PROG_LIBTOOL
--- mjpegtools-1.6.0/lavtools/Makefile.am.orig	Sat Dec 21 19:50:00 2002
+++ mjpegtools-1.6.0/lavtools/Makefile.am	Sat Dec 21 19:52:41 2002
@@ -115,8 +115,8 @@
 	liblavrec.la liblavfile.la liblavjpeg.la \
 	@LIBGETOPT_LIB@ $(PTHREAD_LIBS)
 
-testrec_SOURCES = testrec.c  audiolib.c  $(lav_headers)
-testrec_LDADD = $(PTHREAD_LIBS) $(top_builddir)/utils/libmjpegutils.la
+testrec_SOURCES = testrec.c  $(lav_headers)
+testrec_LDADD = $(PTHREAD_LIBS) $(top_builddir)/utils/libmjpegutils.la audiolib.lo
 
 lav2yuv_SOURCES = lav2yuv.c lav_common.c $(lav_headers)
 lav2yuv_LDADD = $(top_builddir)/utils/libmjpegutils.la \
--- mjpegtools-1.6.0/yuvdenoise/Makefile.am.orig	Sat Dec 21 20:28:59 2002
+++ mjpegtools-1.6.0/yuvdenoise/Makefile.am	Sat Dec 21 20:59:20 2002
@@ -3,7 +3,7 @@
 EXTRA_DIST = .cvsignore
 MAINTAINERCLEANFILES = Makefile.in
 INCLUDES = -I $(top_srcdir)/utils -I $(top_srcdir)
-AM_CFLAGS = -funroll-all-loops -finline-functions -ffast-math -frerun-cse-after-loop -frerun-loop-opt -fexpensive-optimizations -fmove-all-movables -Werror
+AM_CFLAGS = @CFLAGS@ -funroll-all-loops -finline-functions -ffast-math -frerun-cse-after-loop -frerun-loop-opt -fexpensive-optimizations -fmove-all-movables -Werror
 
 bin_PROGRAMS       = yuvdenoise
 yuvdenoise_SOURCES = main.c deinterlace.c deinterlace.h denoise.c denoise.h motion.c motion.h global.h
